apps/wappush/js/messagedb.js

```diff
@@ -76,18 +102,27 @@ var MessageDB = (function() {
 
       req.onsuccess = function mdb_opened(event) {
         db = event.target.result;
-        resolve(db);
+        mdb_deleteViewedMessages().then(() => resolve(db));
       };
       req.onerror = function mdb_openError(event) {
         reject(event.target.error);
       };
       req.onupgradeneeded = function mdb_upgradeNeeded(event) {
```

Comment:
maybe not in this patch, but as soon as we have more than 3 different versions (0, 1, 2) I think it would be better to move this function to a separate function.